### PR TITLE
[Feat](Nereids) add split_part function in fe constant folding

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
split_part('2024-04-17,2024-04-17', ',', 1) ==> '2024-04-17' when fold constant by fe

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by zclllyybb — COMMENTED** [review]
maybe we should have more case to ensure it have same behaviour or fallback to BE of corner cases. like empty text/split strings, invalid index...

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/ExecutableFunctions.java [hunk 1]
```diff
@@ -114,6 +114,11 @@ public static Expression appendTrailingIfCharAbsent(StringLikeLiteral literal, S
                 : new VarcharLiteral(literal.getValue() + chr.getValue());
     }
 
+    @ExecFunction(name = "split_part", argTypes = {"VARCHAR", "VARCHAR", "INT"}, returnType = "VARCHAR")
+    public static Expression splitPart(StringLikeLiteral literal, StringLikeLiteral chr, IntegerLiteral number) {
+        return new VarcharLiteral(literal.getValue().split(chr.getValue())[number.getValue() - 1]);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/ExecutableFunctions.java:119]: should check index out of bound

**Review by morrySnow — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/ExecutableFunctions.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/ExecutableFunctions.java:122]: done

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/ExecutableFunctions.java (hunk 1 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/ExecutableFunctions.java:122]: I think you should not split twice

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Pull request closed without merging**